### ISSUE TITLE
`FullyQualifiedStrictTypesFixer` adds imports before PHP opening tag

### ISSUE THREAD

**Issue by sy-records** [PHP-CS-Fixer/PHP-CS-Fixer#7950]
### Problem description

Fixing bin/hyperf.php produces the following error

https://github.com/hyperf/hyperf-skeleton/blob/master/bin/hyperf.php

```bash

Files that were not fixed due to errors reported during fixing:
   1) /bin/hyperf.php


[TypeError]          
        Illegal offset type  
                             

      PhpCsFixer\Fixer\Semicolon\MultilineWhitespaceBeforeSemicolonsFixer->findWhitespaceBeforeFirstCall()
        in /vendor/friendsofphp/php-cs-fixer/src/Fixer/Semicolon/MultilineWhitespaceBeforeSemicolonsFixer.php at line 112
      PhpCsFixer\Fixer\Semicolon\MultilineWhitespaceBeforeSemicolonsFixer->applyFix()
        in /vendor/friendsofphp/php-cs-fixer/src/AbstractFixer.php at line 75
      PhpCsFixer\AbstractFixer->fix()
        in /vendor/friendsofphp/php-cs-fixer/src/Runner/Runner.php at line 173
      PhpCsFixer\Runner\Runner->fixFile()
        in /vendor/friendsofphp/php-cs-fixer/src/Runner/Runner.php at line 114
      PhpCsFixer\Runner\Runner->fix()
        in /vendor/friendsofphp/php-cs-fixer/src/Console/Command/FixCommand.php at line 305
      PhpCsFixer\Console\Command\FixCommand->execute()
        in /vendor/symfony/console/Command/Command.php at line 326
      [ ... ]
Script @php vendor/bin/php-cs-fixer fix $1 handling the cs-fix event returned with error code 64

```

### Minimal reproducer

```php
return (new PhpCsFixer\Config())
    ->setRiskyAllowed(true)
    ->setRules([
        '@PhpCsFixer' => true,
    ])
    ->setUsingCache(false);
```

```php
#!/usr/bin/env php
<?php

(new Composer\Autoload\ClassLoader())->getPrefixes();
```

My tests have something to do with it.

```diff
-#!/usr/bin/env php
```

### Fixer version

3.53.0

### PHP version

8.2.10

### How do you run Fixer?

Composer package

### Contribution Checks

- [X] I have verified if this problem was already reported
- [X] I am familiar with ["Feature or bug?"](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/feature-or-bug.rst)

**Comment by Wirone** [comment]
Hi @sy-records, thank you for the report! How did you like our new issue template?

Anyway, reproducible also on PHP 8.3, with slightly different error.

```
$ php-cs-fixer check --verbose -vvv --diff --rules=@PhpCsFixer reproducer.php

PHP CS Fixer 3.53.1-DEV 15 Keys by Fabien Potencier, Dariusz Ruminski and contributors.
PHP runtime: 8.3.4
Loaded config default from "[...]/.php-cs-fixer.php".
Paths from configuration file have been overridden by paths provided as command arguments.
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

Found 0 of 1 files that can be fixed in 0.008 seconds, 18.000 MB memory used

Files that were not fixed due to errors reported during fixing:
   1) reproducer.php
                                                            
        [TypeError]                                         
        Cannot access offset of type null on SplFixedArray                         

      PhpCsFixer\Fixer\Semicolon\MultilineWhitespaceBeforeSemicolonsFixer->findWhitespaceBeforeFirstCall()
        in src/Fixer/Semicolon/MultilineWhitespaceBeforeSemicolonsFixer.php at line 112
      PhpCsFixer\Fixer\Semicolon\MultilineWhitespaceBeforeSemicolonsFixer->applyFix()
        in src/AbstractFixer.php at line 75
      PhpCsFixer\AbstractFixer->fix()
        in src/Runner/Runner.php at line 173
      ...
```

The error comes from `multiline_whitespace_before_semicolons`, but in fact it's caused by `fully_qualified_strict_types` which adds an import in the wrong place:

<img width="901" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/f20e0446-26ec-4339-b4ec-50ce256e9d1c">

This leads the `while` loop in `MultilineWhitespaceBeforeSemicolonsFixer::findWhitespaceBeforeFirstCall()` rewinding to token 0 (shebang), and then it throws an exception.

The actual fix should go to `FullyQualifiedStrictTypesFixer` somewhere [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/0e9f115f5ebdd2b71a82562f3d844ac30a1bf2ed/src/Fixer/Import/FullyQualifiedStrictTypesFixer.php#L338-L349), because `$atIndex` is determined wrong.

**Comment by sy-records** [comment]
Thank you for testing @Wirone 

> How did you like our new issue template?

It's great. I wasn't able to reproduce it via the CLI when I reproduced it though, so I provided the code block.
